compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -138,7 +138,7 @@ public DocumentId documentId(Path file) {
                 String moduleDirName;
                 if (moduleId.moduleName().contains(DOT)) {
                     moduleDirName = moduleId.moduleName()
-                            .split(this.currentPackage().packageName().toString() + DOT)[1];
+                            .split(this.currentPackage().packageName().toString() + "\\.")[1];
```

Comment:
Shall we add some unit tests for documentId

Reply:
Added a unit tets